src/organisation/components/pba-numbers-form/pba-numbers-form.component.ts

```diff
@@ -22,48 +25,82 @@ export class PbaNumbersFormComponent implements OnInit {
   };
 
   @Input()
-  public updatePbaNumbers: UpdatePbaNumbers;
+  public organisation: Organisation;
 
-  constructor(private readonly fb: FormBuilder) { }
+  constructor(
+    private readonly router: Router,
+    private readonly orgStore: Store<fromStore.OrganisationState>,
+    private readonly fb: FormBuilder) { }
 
-  ngOnInit() {
+  public ngOnInit(): void {
+    this.getOrganisationDetailsFromStore();
     this.initialiseForm();
   }
 
   get pbaNumbers(): FormArray {
     return this.pbaFormGroup.get('pbaNumbers') as FormArray;
   }
 
+  get hasPendingChanges(): boolean {
+    return this.organisation.pendingAddPaymentAccount.length > 0 || this.organisation.pendingRemovePaymentAccount.length > 0;
+  }
+
+  /**
+   * Current PBA Numbers contain existing and pending additions, minus pending removals
+   */
+  get currentPaymentAccounts(): string[] {
+    const currentPbas = this.organisation.paymentAccount
+      .filter(pba => this.organisation.pendingRemovePaymentAccount.indexOf(pba) === -1);
+
+    return currentPbas;
+  }
+
+  get noAccountsAdded(): boolean {
+    return this.organisation.pendingAddPaymentAccount.length === 0 && this.currentPaymentAccounts.length === 0;
+  }
+
   public onAddNewBtnClicked(): void {
     this.pbaNumbers.push(this.newPbaNumber());
   }
 
   public onRemoveNewPbaNumberClicked(i: number): void {
+    const pbaValue = this.pbaNumbers.at(i).value.pbaNumber;
+    const pendingAddPaymentAccount = this.organisation.pendingAddPaymentAccount.filter((pba: string) => pba !== pbaValue);
+
+    this.orgStore.dispatch(new fromStore.UpdateOrganisationPendingAddPbas(pendingAddPaymentAccount));
     this.pbaNumbers.removeAt(i);
+    this.refreshValidation();
   }
 
-  public onRemoveExistingPaymentByAccountNumberClicked(paymentByAccountNumber: string): void {
-    this.updatePbaNumbers.addPbaNumberToPendingRemove(paymentByAccountNumber);
+  public onClickContinue(): void {
+    if (this.hasPendingChanges) {
+      return this.onSubmit();
+    }
 
-    this.pbaNumbers.controls.forEach((c: AbstractControl) => {
-      const pbaControl = c.get('pbaNumber');
-      pbaControl.setValidators(this.getPbaNumberValidators());
-      pbaControl.updateValueAndValidity();
-    });
+    this.summaryErrors = {
+      isFromValid: false,
+      header: 'There is a problem',
+      items: [
+        {
+          id: 'new-pba-form',
+          message: 'Add or remove a PBA account'
```

Comment:
Same query as above for "PBA account".

Reply:
Resolving comment since it's been confirmed the UX team were happy with "PBA account".